tests/Integration/Adapter/ConfigurationTest.php

```diff
@@ -0,0 +1,230 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Integration\Adapter;
+
+use Doctrine\ORM\EntityManager;
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShopBundle\Entity\Shop;
+use PrestaShopBundle\Entity\ShopGroup;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class ConfigurationTest extends KernelTestCase
+{
+    /**
+     * @var Configuration|null
+     */
+    private $configuration;
+
+    /**
+     * @var EntityManager
+     */
+    private $entityManager;
+
+    protected function setUp()
+    {
+        self::bootKernel();
+
+        $container = self::$kernel->getContainer();
+        $this->entityManager = $container->get('doctrine.orm.entity_manager');
+        $this->configuration = $container->get('prestashop.adapter.legacy.configuration');
+
+        $this->initMultistore();
+    }
+
+    /**
+     * @param array $setParams
+     * @param array $getParams
+     * @param $expectedResult
+     * @param bool $isLastTest
+     *
+     * @dataProvider getProvider
+     */
+    public function testGet(array $setParams, array $getParams, $expectedResult, bool $isLastTest): void
+    {
+        foreach ($setParams as $values) {
+            $this->configuration->set($values['key'], $values['value'], $values['shopConstraint']);
+        }
+        $result = $this->configuration->get($getParams['key'], $getParams['default'], $getParams['shopConstraint']);
+
+        $this->assertEquals($expectedResult, $result);
+
+        if ($isLastTest) {
+            $this->cleanDb();
+        }
+    }
+
+    /**
+     * @return array[]
+     */
+    public function getProvider(): array
```

Comment:
The way the provider and the test data are built is a bit confusing and hard to understand
Maybe using some `yield` to separate each test clearly would make it more readable

And you could split the tests with `isStrict` in a dedicated test function (with its dedicated provider as well)